src/Structures/GridD.cs

```diff
@@ -0,0 +1,172 @@
+using Fusee.Math.Core;
+using System;
+using System.Collections.Generic;
+
+namespace Fusee.Structures
+{
+    /// <summary>
+    /// A data structure that represents a three dimensional grid.
+    /// </summary>
+    /// <typeparam name="P">The type of the payload.</typeparam>
+    public abstract class GridD<P>
+    {
+        /// <summary>
+        /// All grid cells as three dimensional array.
+        /// </summary>
+        public GridCellD<P>[,,] GridCells { get; set; }
+
+        /// <summary>
+        /// The number of grid cells in each dimension.
+        /// </summary>
+        public readonly int3 NumberOfGridCells;
+
+        /// <summary>
+        /// The size of a grid cell.
+        /// </summary>
+        public readonly double3 CellSize;
+
+        /// <summary>
+        /// The center of the grid.
+        /// </summary>
+        public readonly double3 Center;
+
+        /// <summary>
+        /// The size of the grid.
+        /// </summary>
+        public double3 Size
+        {
+            get
+            {
+                if (_size == null || _size == double3.Zero)
+                    return new double3(CellSize.x * NumberOfGridCells.x, CellSize.y * NumberOfGridCells.y, CellSize.z * NumberOfGridCells.z);
+                else return _size;
+            }
+        }
+        private double3 _size;
+
+        /// <summary>
+        /// Creates a new instance of type GridD.
+        /// </summary>
+        /// <param name="center">The center of the grid.</param>
+        /// <param name="size">The size of the grid.</param>
+        /// <param name="noOfCellsX">Number of cells in x direction.</param>
+        /// <param name="noOfCellsY">Number of cells in y direction.</param>
+        /// <param name="noOfCellsZ">Number of cells in z direction.</param>
+        public GridD(double3 center, double3 size, int noOfCellsX, int noOfCellsY, int noOfCellsZ)
+        {
+            _size = size;
+            Center = center;
+            NumberOfGridCells = new int3(noOfCellsX, noOfCellsY, noOfCellsZ);
+            CellSize = new double3(size.x / noOfCellsX, size.y / noOfCellsY, size.z / noOfCellsZ);
+            GridCells = new GridCellD<P>[noOfCellsX, noOfCellsY, noOfCellsZ];
+        }
+
+        /// <summary>
+        /// Creates all cells for this grid without any payload.
+        /// </summary>
+        public void CreateCells()
```

Comment:
This is intentional because there also is `CreateCellForItem` one can use when they do not want to create a "full" Grid but instatiate cells on demand. I think its better to leave it to the user how and which cells he wants to create.